base/reducedim.jl

```diff
@@ -165,6 +164,41 @@ for (f1, f2, initval, typeextreme) in ((:min, :max, :Inf, :typemax), (:max, :min
         end
     end
 end
+
+function reducedim_init(f::ExtremaMap, op::typeof(_extrema_rf), A::AbstractArray, region)
+    # First compute the reduce indices. This will throw an ArgumentError
+    # if any region is invalid
+    ri = reduced_indices(A, region)
+
+    # Next, throw if reduction is over a region with length zero
+    any(i -> isempty(axes(A, i)), region) && _empty_reduce_error()
+
+    # Make a view of the first slice of the region
+    A1 = view(A, ri...)
+
+    isempty(A1) && return map(f, A1)
+    # use the max/min of the first slice as initial value for non-empty cases
+    v0 = reverse(mapreduce(f, op, A1)) # turn minmax to maxmin
+
+    T = _realtype(f.f, promote_union(eltype(A)))
+    Tr = v0[1] isa T && v0[2] isa T ? NTuple{2,T} : typeof(v0)
+
+    # but NaNs and missing need to be avoided as initial values
+    if v0[1] isa Number && isnan(v0[1])
+        v0 = oftype(v0[1], Inf), oftype(v0[2], -Inf)
+    elseif isunordered(v0[1])
+        # v0 is missing or a third-party unordered value
+        T1, T2 = Tr.parameters[1], Tr.parameters[2]
```

Comment:
It's better to avoid touching the internal presentation of `DataType`. Why not

```suggestion
        T1 = typeof(Tr[1])
        T1 = typeof(Tr[2])
```

?

Reply:
I guess `getindex` is undefined for `Tuple`?
```julia
Tuple{Int,Int}[1]
ERROR: MethodError: Cannot `convert` an object of type Int64 to an object of type Tuple{Int64, Int64}
```
I'll split them at the beginning to avoid accessing `parameters`

Reply:
Right, I meant something like

```suggestion
        T1 = typeof(v0[1])
        T1 = typeof(v0[2])
```

?

Reply:
For this case, both `T1` and `T2` are `Missing` . I guess the best choice is the following:
```julia
    Tmin = v0[1] isa T ? T : typeof(v0[1])
    Tmax = v0[2] isa T ? T : typeof(v0[2])
    if v0[1] isa Number && isnan(v0[1])
        v0 = oftype(v0[1], Inf), oftype(v0[2], -Inf)
    elseif isunordered(v0[1])
        v0 = typemax(nonmissingtype(Tmin)), typemin(nonmissingtype(Tmax))
    end
```

Reply:
Yes, that looks like the closest to how `minimum`/`maximum` handles this